web/modules/custom/weather_block/weather_block.schema.yml

```diff
@@ -0,0 +1,7 @@
+block.settings.hello_block:
+  type: block_settings
+  label: 'Weather block settings'
+  mapping:
+    hello_block_name:
+      type: string
+      label: 'Weather Block'
```

Comment:
If you are not willing to provide the correct schema of configuration, then do not create it (the schema.yml) at all.
It is not required